### PR TITLE
fix:  sanity test failures in create volume

### LINKED ISSUE #27: fix sanity test failures
**What happened**:

```
Summarizing 5 Failures:

[Fail] Controller Service CreateVolume [It] should not fail when requesting to create a volume with already existing name and same capacity. 
/home/priyanshu/work/src/github.com/csi-driver/azurefile-csi-driver/vendor/github.com/kubernetes-csi/csi-test/pkg/sanity/controller.go:371

[Fail] Controller Service CreateVolume [It] should fail when requesting to create a volume with already existing name and different capacity. 
/home/priyanshu/work/src/github.com/csi-driver/azurefile-csi-driver/vendor/github.com/kubernetes-csi/csi-test/pkg/sanity/controller.go:442

[Fail] Controller Service ValidateVolumeCapabilities [It] should fail when the requested volume does not exist 
/home/priyanshu/work/src/github.com/csi-driver/azurefile-csi-driver/vendor/github.com/kubernetes-csi/csi-test/pkg/sanity/controller.go:705

[Fail] Node Service [It] should work 
/home/priyanshu/work/src/github.com/csi-driver/azurefile-csi-driver/vendor/github.com/kubernetes-csi/csi-test/pkg/sanity/node.go:464

Ran 26 of 57 Specs in 77.639 seconds
FAIL! -- 21 Passed | 5 Failed | 0 Pending | 31 Skipped
```

following issue has been fixed:
```
[Fail] Controller Service DeleteVolume [It] should succeed when an invalid volume id is used 
/home/priyanshu/work/src/github.com/csi-driver/azurefile-csi-driver/vendor/github.com/kubernetes-csi/csi-test/pkg/sanity/controller.go:543
```
**What you expected to happen**:

**How to reproduce it**:
run sanity test: https://github.com/kubernetes-sigs/azurefile-csi-driver/tree/master/test/sanity

**Anything else we need to know?**:

**Environment**:
- CSI Driver version:
- Kubernetes version (use `kubectl version`):
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by ZeroMagic**
When we do sanity test, "create a volume with already existing name and different capacity" can pass.

Signed-off-by: ZeroMagic <anthonyliu@zju.edu.cn>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind failing-test

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #27 

**Special notes for your reviewer**:

**Release note**:
```
fix:  sanity test failures in create volume
```

### CODE REVIEW

File: pkg/azurefile/azurefile.go [hunk 1]
```diff
@@ -108,28 +115,51 @@ func (d *Driver) Run(endpoint string) {
 	s.Wait()
 }
 
-func (d *Driver) checkFileShareExists(accountName, resourceGroup, name string) (bool, error) {
+func (d *Driver) checkFileShareExists(accountName, resourceGroup, name string, accountKey *string) (bool, error) {
```
Comment (andyzhangx) [inline pkg/azurefile/azurefile.go]: I think you could let `checkFileShareExists` func return key directly

File: pkg/azurefile/azurefile.go [hunk 2]
```diff
@@ -108,28 +115,51 @@ func (d *Driver) Run(endpoint string) {
...
-	fileClient, err := d.getFileSvcClient(accountName, accountKey)
+	fileClient, err := d.getFileSvcClient(accountName, key)
 	if err != nil {
-		return false, fmt.Errorf("error creating azure client: %v", err)
+		return false, err
 	}
 	return fileClient.GetShareReference(name).Exists()
 }
 
+func (d *Driver) checkCapacity(accountName, accountKey, fileShareName string, requestGiB int) error {
```
Comment (andyzhangx) [inline pkg/azurefile/azurefile.go]: checkFileShareCapacity

File: pkg/azurefile/azurefile.go [hunk 3]
```diff
@@ -108,28 +115,51 @@ func (d *Driver) Run(endpoint string) {
...
-func (d *Driver) checkFileShareExists(accountName, resourceGroup, name string) (bool, error) {
+func (d *Driver) checkFileShareExists(accountName, resourceGroup, name string, accountKey *string) (bool, error) {
 	// find the access key with this account
-	accountKey, err := d.cloud.GetStorageAccesskey(accountName, resourceGroup)
+	key, err := d.cloud.GetStorageAccesskey(accountName, resourceGroup)
 	if err != nil {
 		// Because the account does not exist in the resource group, the file share could not exist.
 		// In the next step, we will create a temp account to create file share.
 		// So here we return nil, and warn the user.
 		klog.Warningf("could not get storage key for storage account %s: %v", accountName, err)
```
Comment (andyzhangx) [inline pkg/azurefile/azurefile.go]: pls remove this comment, could not find the storage account key is a common scenario when user does not input `accountName`

File: pkg/azurefile/azurefile.go [hunk 4]
```diff
@@ -45,6 +48,10 @@ const (
 	// See https://docs.microsoft.com/en-us/rest/api/storageservices/naming-and-referencing-shares--directories--files--and-metadata#share-names
 	fileShareNameMinLength = 3
 	fileShareNameMaxLength = 63
+
+	// See https://docs.microsoft.com/en-us/rest/api/storageservices/create-share#request-headers
+	quotaMax = 5120
+	quotaMin = 0
```
Comment (andyzhangx) [inline pkg/azurefile/azurefile.go]: actually `quotaMin` is 1 (GB)

File: pkg/azurefile/azurefile.go [hunk 5]
```diff
@@ -108,28 +115,51 @@ func (d *Driver) Run(endpoint string) {
...
+	fileClient, err := d.getFileSvcClient(accountName, accountKey)
+	if err != nil {
+		return err
+	}
+	resp, err := fileClient.ListShares(azs.ListSharesParameters{Prefix: fileShareName})
+	if err != nil {
+		return fmt.Errorf("error listing file shares: %v", err)
+	}
+	for _, share := range resp.Shares {
+		// When the requestGiB is null, "CreateShare" will set the quota to max value "5120".
```
Comment (andyzhangx) [inline pkg/azurefile/azurefile.go]: a little confused about the logic here:
 - `requestGiB` is set by user, so if `requestGiB` is 0, then it should return error, right?

File: pkg/azurefile/controllerserver.go [hunk 6]
```diff
@@ -88,18 +89,19 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 		fileShareName = getValidFileShareName(name)
 	}
 
-	if exists, err := d.checkFileShareExists(account, resourceGroup, fileShareName); err != nil {
+	if exists, err := d.checkFileShareExists(account, resourceGroup, fileShareName, accountKey); err != nil {
 		return nil, err
 	} else if exists {
 		volumeID = fmt.Sprintf(volumeIDTemplate, resourceGroup, account, fileShareName)
```
Comment (andyzhangx) [inline pkg/azurefile/controllerserver.go]: I think the `checkFileShareCapacity` process only happens when file share exists, so what about merge `checkFileShareCapacity` and `checkFileShareExists` into one function?

**Comment by ZeroMagic** [comment]
I have modified the code related requestGiB. We can go ahead to review.

File: pkg/azurefile/azurefile.go [hunk 7]
```diff
@@ -45,6 +48,10 @@ const (
 	// See https://docs.microsoft.com/en-us/rest/api/storageservices/naming-and-referencing-shares--directories--files--and-metadata#share-names
 	fileShareNameMinLength = 3
 	fileShareNameMaxLength = 63
+
+	// Minimum size of Azure Premium Files is 100GiB
+	// See https://docs.microsoft.com/en-us/azure/storage/files/storage-files-planning#provisioned-shares
+	quotaDefault = 100
```
Comment (andyzhangx) [inline pkg/azurefile/azurefile.go]: defaultAzureFileQuota

Comment (ZeroMagic) [reply pkg/azurefile/azurefile.go]: done

**Pull request merged**